rspace/src/main/scala/coop/rchain/rspace/history/LMDBTrieStore.scala

```diff
@@ -48,6 +51,22 @@ class LMDBTrieStore[K, V] private (val env: Env[ByteBuffer], _dbTrie: Dbi[ByteBu
       Codec[Trie[K, V]].decode(BitVector(buffer)).map(_.value).get
     }
   }
+
+  private[rspace] def toMap: Map[Blake2b256Hash, Trie[K, V]] =
+    withTxn(createTxnRead()) { txn =>
+      val keyRange: KeyRange[ByteBuffer] = KeyRange.all()
+      withResource(_dbTrie.iterate(txn, keyRange)) { (it: CursorIterator[ByteBuffer]) =>
+        it.asScala.foldLeft(Map.empty[Blake2b256Hash, Trie[K, V]]) {
+          (map: Map[Blake2b256Hash, Trie[K, V]], x: CursorIterator.KeyVal[ByteBuffer]) =>
+            val key   = Codec[Blake2b256Hash].decode(BitVector(x.key())).map(_.value).get
+            val value = Codec[Trie[K, V]].decode(BitVector(x.`val`())).map(_.value).get
```

Comment:
Are backticks needed around `val` here?

Reply:
yes, because `val` is a reserved word.  i have no control here, this is part of the way scala handles the `lmdbjava` library.